.github/workflows/on_pull_request.yaml

```diff
@@ -0,0 +1,17 @@
+name: On Pull Request
+
+on:
+  pull_request:
+
+jobs:
+
+  on-pull-request:
+    name: Get ROCKs modified and build-scan-test them
+    uses: canonical/charmed-kubeflow-workflows/.github/workflows/get-rocks-modified-and-build-scan-test-publish.yaml@main
+    permissions:
+      pull-requests: read
+    secrets: inherit
+    with:
+      microk8s-channel: 1.24/stable
```

Comment:
I wonder if we should use a newer version of microk8s and juju. I believe the workflow doesn't really care on the version, but since all our repos now use newer ones, we probably want that here as well.

Reply:
Its what we use in template https://github.com/canonical/charmed-kubeflow-workflows/blob/main/templates/on-push-rock-template.yaml

Reply:
We also use 1.24 in some of our [other repos](https://github.com/canonical/filebrowser-rock/blob/main/.github/workflows/on_push.yaml), but I dont know why we'd use 1.24.  We should change them all at once though, so I vote we approve here but then change everything at once.  wdyt?

Reply:
I did the bump already for [kubeflow-rocks](https://github.com/canonical/kubeflow-rocks/blob/main/.github/workflows/on_pull_request.yaml#L15C1-L18C39) and it is working. Actually, I think it will be worth having the latest versions because we run tests that deploy the charm(s) and what not, so the k8s version may be relevant. I'm fine if we do a cannon run for changing all repos later, though.

Reply:
Alrighty. I am adding a note to discuss with the team how we'll do the update to other repositories.